app/api/templates/utils.ts

```diff
@@ -135,3 +136,12 @@ export function getDeletedProperties(
     .filter(notIncludedIn(flattenProperties(newProperties)))
     .map(property => property[prop]);
 }
+
+export function getDeletedCommonProperties(
```

Comment:
It seems to me that the `notIncludedIn` function tests inclusion by id. In this case, the commonProperty does not get deleted, just the name is changed, it has the same id.

I agree that the design of the function is a bit wonky, since at the moment, this is the only commonProperty where we care about this, so putting it into a separate function might be too much. In this case, I went with readability, since this way it follows the same pattern as with the properties. Probably the naming could've been different. We can discuss this.

Reply:
Makes total sense, thanks for clarifying! Maybe yes, we can try to figure out a better naming. In fact, common properties cannot be deleted (which also isn't the case for your algorithm). I think something around `renamed` would be better. Or even better, let's make it refer right to the title, since it's the only thing it's acting on. Yeah, let's have a chat!